src/components/Shifting/ShiftingList.tsx

```diff
@@ -62,42 +72,177 @@ export default function ListView() {
+            </dt>
+
+            {careConfig.wartimeShifting && (
+              <dt
+                title={t("shifting_approving_facility")}
+                className="flex items-center text-left text-sm font-medium leading-5 text-secondary-500"
+              >
+                <CareIcon icon="l-user-check" className="mr-2" />
+                <dd className="text-sm font-bold leading-5 text-secondary-900">
+                  {(shift.shifting_approving_facility_object || {}).name}
+                </dd>
+              </dt>
+            )}
+
+            <dt
+              title={t("assigned_facility")}
+              className="flex items-center text-left text-sm font-medium leading-5 text-secondary-500"
+            >
+              <CareIcon icon="l-plane-arrival" className="mr-2" />
+              <dd className="text-sm font-bold leading-5 text-secondary-900">
+                {shift.assigned_facility_external ||
+                  shift.assigned_facility_object?.name ||
+                  t("yet_to_be_decided")}
+              </dd>
+            </dt>
+          </div>
+          <div className="col-span-1 mt-2 flex flex-col text-left">
+            <ButtonV2
+              onClick={(_) => navigate(`/shifting/${shift.external_id}`)}
+              variant="secondary"
+              border
+              className="w-full"
+            >
+              <CareIcon icon="l-eye" className="mr-2" /> {t("all_details")}
+            </ButtonV2>
+            {shift.status === "COMPLETED" && shift.assigned_facility && (
+              <div className="mt-2">
+                <ButtonV2
+                  className="w-full"
+                  disabled={
+                    !shift.patient_object.allow_transfer ||
+                    !(
+                      ["DistrictAdmin", "StateAdmin"].includes(
+                        authUser.user_type,
+                      ) ||
+                      authUser.home_facility_object?.id ===
+                        shift.assigned_facility
+                    )
+                  }
+                  onClick={() =>
+                    setModalFor({
+                      externalId: shift.external_id,
+                      loading: false,
+                    })
+                  }
+                >
+                  {t("transfer_to_receiving_facility")}
+                </ButtonV2>
+                <ConfirmDialog
+                  title={t("confirm_transfer_complete")}
+                  description={t("mark_transfer_complete_confirmation")}
+                  action="Confirm"
+                  show={modalFor.externalId === shift.external_id} // Check the externalId here
+                  onClose={() =>
+                    setModalFor({ externalId: undefined, loading: false })
+                  }
+                  // show={modalFor === shift.external_id}
+                  // onClose={() =>
+                  //   setModalFor({ externalId: undefined, loading: false })
+                  // }
```

Comment:
```suggestion
```

Reply:
it will raise the error bcoz we pass undefined for external Id in close modal so we need to update models

Reply:
what does removing comments have anything to do with causing errors?